### PR TITLE
[macOS] Move Ruby and Go versions to toolset

### PR DESCRIPTION
**Pull request by dibir-magomedsaygitov**
# Description
In scope of this PR we are moving default Ruby and Go versions to toolset.

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/macos/toolsets/toolset-10.15.json [hunk 1]
```diff
@@ -304,5 +304,11 @@
         "xcpretty",
         "bundler",
         "fastlane"
-    ]
+    ],
+    "default_ruby": {
```
Comment (al-cheb) [inline images/macos/toolsets/toolset-10.15.json:308]: I think it should be called like `ruby` and contains `rubygems` and `default` like `java` section

**Review by maxim-lobanov — DISMISSED** [review]

**Review by al-cheb — DISMISSED** [review]

File: images/macos/toolsets/toolset-11.json [hunk 2]
```diff
@@ -233,12 +233,18 @@
...
+            "xcode-install",
+            "cocoapods",
+            "nomad-cli",
+            "xcpretty",
+            "bundler",
+            "fastlane"
+        ]
+    },
+    "go": {
+        "default": "1.15"
```
Comment (MaksimZhukov) [inline images/macos/toolsets/toolset-11.json:248]: I believe we should remove `go@1.15` from the `common_packages` section as we did for others toolsets

Comment (dibir-magomedsaygitov) [reply images/macos/toolsets/toolset-11.json:247]: thank you, fixed

**Review by MaksimZhukov — APPROVED** [review]

**Review by al-cheb — APPROVED** [review]

File: images/macos/provision/core/golang.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash -e -o pipefail
+source ~/utils/utils.sh
+
+DEFAULT_GO_VERSION=$(get_toolset_value '.go.default')
+echo "Installing Go..."
+brew_smart_install "go@${DEFAULT_GO_VERSION}"
+
+# Create symlinks for Go 1.15 to preserve backward compatibility
+ln -sf $(brew --prefix go@${DEFAULT_GO_VERSION})/bin/* /usr/local/bin/
```
Comment (miketimofeev) [inline images/macos/provision/core/golang.sh:9]: the problem is that we need to create a symlink not only for 1.15 but for any non-latest go version. We have to either change a comment here or create a condition to check whether the installed version is the same as `brew install go` or not

**Review by maxim-lobanov — DISMISSED** [review]

**Review by MaksimZhukov — DISMISSED** [review]

File: images/macos/provision/core/golang.sh [hunk 4]
```diff
@@ -5,7 +5,7 @@ DEFAULT_GO_VERSION=$(get_toolset_value '.go.default')
 echo "Installing Go..."
 brew_smart_install "go@${DEFAULT_GO_VERSION}"
 
-# Create symlinks for Go 1.15 to preserve backward compatibility
+#Create symlinks to preserve backward compatibility. Symlinks are not created when non-latest go is being installed
```
Comment (miketimofeev) [inline images/macos/provision/core/golang.sh:8]:
```suggestion
# Create symlinks to preserve backward compatibility. Symlinks are not created when non-latest go is being installed
```

**Review by MaksimZhukov — APPROVED** [review]

**Pull request merged**